src/components/tabs/tabs.js

```diff
@@ -0,0 +1,282 @@
+  if (typeof window.matchMedia === 'function') {
+    this.setupResponsiveChecks()
+  } else {
+    this.setup()
+  }
+}
+
+Tabs.prototype.setupResponsiveChecks = function () {
+  this.mql = window.matchMedia('(min-width: 40.0625em)')
+  this.mql.addListener(this.checkMode.bind(this))
+  this.checkMode(this.mql)
+}
+
+Tabs.prototype.checkMode = function (mql) {
+  if (this.mql.matches) {
+    this.setup()
+  } else {
+    this.teardown()
+  }
+}
+
+Tabs.prototype.setup = function () {
+  var $module = this.$module
+  var $tabs = this.$tabs
+  var $tabList = $module.querySelector('.govuk-tabs__list')
+  var $tabListItems = $module.querySelectorAll('.govuk-tabs__list-item')
+
+  if (!$tabs || !$tabList || !$tabListItems) {
+    return
+  }
+
+  $tabList.setAttribute('role', 'tablist')
+
+  nodeListForEach($tabListItems, function ($item) {
+    $item.setAttribute('role', 'presentation')
+  })
+
+  nodeListForEach($tabs, function ($tab) {
+    // Set HTML attributes
+    this.setAttributes($tab)
+
+    // Handle events
+    $tab.boundTabClick = this.onTabClick.bind(this)
+    $tab.boundTabKeydown = this.onTabKeydown.bind(this)
+    $tab.addEventListener('click', $tab.boundTabClick, true)
+    $tab.addEventListener('keydown', $tab.boundTabKeydown, true)
+
+    // Remove old active panels
+    this.unhighlightTab($tab)
+    this.hidePanel($tab)
+  }.bind(this))
+
+  // If there's a tab that matches the hash
+  var $activeTab = this.getTab(window.location.hash)
+  if ($activeTab) {
+    this.highlightTab($activeTab)
+    this.showPanel($activeTab)
+  } else {
+    // or show the first
+    var firstTab = this.$tabs[0]
+    this.highlightTab(firstTab)
+    this.showPanel(firstTab)
+  }
+
+  // Handle hashchange events
+  $module.boundOnHashChange = this.onHashChange.bind(this)
+  window.addEventListener('hashchange', $module.boundOnHashChange, true)
+}
+
+Tabs.prototype.teardown = function () {
+  var $module = this.$module
+  var $tabs = this.$tabs
+  var $tabList = $module.querySelector('.govuk-tabs__list')
+  var $tabListItems = $module.querySelectorAll('.govuk-tabs__list-item')
+
+  if (!$tabs || !$tabList || !$tabListItems) {
+    return
+  }
+
+  $tabList.removeAttribute('role')
+
+  nodeListForEach($tabListItems, function ($item) {
+    $item.removeAttribute('role', 'presentation')
+  })
+
+  nodeListForEach($tabs, function ($tab) {
+    // Remove events
+    $tab.removeEventListener('click', $tab.boundTabClick, true)
+    $tab.removeEventListener('keydown', $tab.boundTabKeydown, true)
+
+    // Unset HTML attributes
+    this.unsetAttributes($tab)
+  }.bind(this))
+
+  // Remove hashchange event handler
+  window.removeEventListener('hashchange', $module.boundOnHashChange, true)
+}
+
+Tabs.prototype.onHashChange = function (e) {
+  var hash = window.location.hash
+  if (!this.hasTab(hash)) {
+    return
+  }
+  if (this.changingHash) {
+    this.changingHash = false
+    return
+  }
+  var $hashTab = this.getTab(window.location.hash)
```

Comment:
Can use `hash` rather than `window.location.hash`.

Would also consider skipping `hasTab` and storing and using the result from `getTab` which would mirror what we do in setup (`var $activeTab = this.getTab(window.location.hash)`)